### DISCUSSION TITLE
[feature proposal] grouping state variants

### DISCUSSION THREAD

**Discussion by uwenayoallain** [tailwindlabs/tailwindcss#4935] [Ideas]
# Grouping state variants

sometimes using state variants can result in using many classes for an element.
Ex: 
```html
  <input class="border border-transparent focus:outline-none focus:ring-2 focus:ring-purple-600
 focus:border-transparent  ...">
```
in the above example, the `focus` state variant is being repeated a lot.
so my proposal is that many `utilities` can be prefixed to a `single` `state` `variant` like:
```html
  <input class="border border-transparent focus:[outline-none ring-2 ring-purple-600 border-transparent] ...">
```
without repeating same state variants;

**Comment by lukewarlow** [comment]
Potentially would need to use () because [] is already used for arbitrary value support. But this would be a massively useful change for writing readable markup.
